### ISSUE TITLE
"invalid literal for int()" during munkiimport 

### ISSUE THREAD

**Issue by bryanheinz** [munki/munki#902]
I was trying to import CloudBerry Backup for macOS (https://www.cloudberrylab.com/backup/desktop/mac.aspx) when I got the "invalid literal for int()" error.

This is the full error:
```
munkiimport mac_CloudBerryLab_CloudBerryBackup_v2.7.0.28.pkg 
Traceback (most recent call last):
  File "/usr/local/munki/munkiimport", line 614, in <module>
    main()
  File "/usr/local/munki/munkiimport", line 384, in main
    pkginfo = pkginfolib.makepkginfo(installer_item, options)
  File "/usr/local/munki/munkilib/admin/pkginfolib.py", line 434, in makepkginfo
    pkginfo = get_catalog_info_from_path(installeritem, options)
  File "/usr/local/munki/munkilib/admin/pkginfolib.py", line 97, in get_catalog_info_from_path
    cataloginfo = pkgutils.getPackageMetaData(pkgpath)
  File "/usr/local/munki/munkilib/pkgutils.py", line 745, in getPackageMetaData
    receiptinfo = getReceiptInfo(pkgitem)
  File "/usr/local/munki/munkilib/pkgutils.py", line 541, in getReceiptInfo
    info = getFlatPackageInfo(pkgname)
  File "/usr/local/munki/munkilib/pkgutils.py", line 368, in getFlatPackageInfo
    infoarray.extend(parsePkgRefs(packageinfoabspath))
  File "/usr/local/munki/munkilib/pkgutils.py", line 268, in parsePkgRefs
    'installKBytes'].value.encode('UTF-8'))
ValueError: invalid literal for int() with base 10: '99407.5615234375'
```

I took a look at line 268 in the pkgutils.py file and reviewed this block:
```
pkginfo['installed_size'] = int(
    payloads[0].attributes[
    'installKBytes'].value.encode('UTF-8'))
```

It looks like it's pulling in a string "float" ("99407.5615234375") and trying to convert it to an integer, which is throwing the error. I tested this out in python and when casting a float to an integer it works fine, but casting a string float into an integer causes the invalid literal error. Casting the string float to a float and it worked.

So, I adjusted the pkgutils.py file locally and tested first casting it to a float, then as an integer and that appears to work. The adjusted code looks like this:
```
pkginfo['installed_size'] = int(float(
    payloads[0].attributes[
    'installKBytes'].value.encode('UTF-8')))
```

After making that change locally I was able to import the CloudBerry package without any errors.

**Comment by gregneagle** [comment]
Interesting. I've never seen a pkg with a float for `installKBytes`. I suspect that's in violation of spec.

**Comment by gregneagle** [comment]
https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/DistributionDefinitionRef/Chapters/Distribution_XML_Ref.html

Defined as an integer.

**Comment by bryanheinz** [comment]
Yeah, it doesn’t surprise me that they’re doing something wrong. The package is pretty rough. I figured that it was a fringe thing, but I wanted to pass along a potential resolution to covering a fringe bad actor. 

I’ll open a support case with them and point out their error.

**Comment by gregneagle** [comment]
Great. I'll also defend against this package authoring mistake in some future release of Munki (but not 3.5.2).

**Comment by gregneagle** [comment]
This will be addressed in the 3.6 release.
